### PR TITLE
Spotlight Next and Previous operations crash fix.

### LINKED ISSUE #91: Spotlight crashing when swiftly selecting next element while previous transition is running
Steps to reproduce:

1. Add a few elements to Spotlight
2. Swiftly invoke `next` action a few times without waiting for the transitions to finish
3. See the crash when you reach the end of the list:

Stacktrace:

```
 java.util.NoSuchElementException: Collection contains no element matching the predicate.
        at com.bumble.appyx.routingsource.spotlight.operation.Next.invoke(Next.kt:52)
        at com.bumble.appyx.routingsource.spotlight.operation.Next.invoke(Next.kt:11)
        at com.bumble.appyx.core.routing.BaseRoutingSource$execute$1.invoke(BaseRoutingSource.kt:114)
        at com.bumble.appyx.core.routing.BaseRoutingSource$execute$1.invoke(BaseRoutingSource.kt:112)
        at com.bumble.appyx.core.routing.BaseRoutingSource.updateState(BaseRoutingSource.kt:106)
        at com.bumble.appyx.core.routing.BaseRoutingSource.execute(BaseRoutingSource.kt:112)
        at com.bumble.appyx.core.routing.BaseRoutingSource.access$execute(BaseRoutingSource.kt:33)
        at com.bumble.appyx.core.routing.BaseRoutingSource$1.invoke(BaseRoutingSource.kt:97)
        at com.bumble.appyx.core.routing.BaseRoutingSource$1.invoke(BaseRoutingSource.kt:97)
        at com.bumble.appyx.core.routing.operationstrategies.ExecuteImmediately.accept(ExecuteImmediately.kt:8)
        at com.bumble.appyx.core.routing.BaseRoutingSource.accept(BaseRoutingSource.kt:101)
        at com.bumble.appyx.routingsource.spotlight.operation.NextKt.next(Next.kt:44)
        at com.bumble.appyx.sandbox.client.spotlight.SpotlightExampleNode$LoadedState$1$2$1$1.invoke(SpotlightExampleNode.kt:160)
        at com.bumble.appyx.sandbox.client.spotlight.SpotlightExampleNode$LoadedState$1$2$1$1.invoke(SpotlightExampleNode.kt:159)
```

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Fixes #91 

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

**Review by LachlanMcKee — COMMENTED** [review]
I don't suppose you were able to write an automation test that reproduced the bug? I.e. switching quickly?

What will our strategy be for avoiding these types of issues in the future?

**Comment by KovalevAndrey** [comment]
> I don't suppose you were able to write an automation test that reproduced the bug? I.e. switching quickly?
> 
> What will our strategy be for avoiding these types of issues in the future?

It could have been covered with a simple unit test which is done in this PR. Having at least unit tests for the critical parts of the code is a good strategy. For some reason we didn't have any tests for Spotlight.

> I.e. switching quickly?

Switching quickly simply means that we update `targetState` while transition is not over, e.g. `fromState=A, toState=B` ->
`fromState=A, toState=C`. This can be simulated with a unit-test so having an instrumentation test is not necessary

**Comment by LachlanMcKee** [comment]
Ah excellent! that's good to know :)

File: core/src/test/java/com/bumble/appyx/navmodel/spotlight/operation/NextTest.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            element = Routing1,
+            fromState = INACTIVE_AFTER,
+            targetState = ACTIVE,
+        )
+        val elements = listOf(firstElement, lastElement)
+        val operation = Next<Routing>()
+
+        val applicable = operation.isApplicable(elements)
+
+        assertEquals(applicable, false)
```
Comment (LachlanMcKee) [inline core/src/test/java/com/bumble/appyx/navmodel/spotlight/operation/NextTest.kt:30]: Do we have an assertFalse/assertTrue we could use instead?

**Review by LachlanMcKee — APPROVED** [review]

**Review by CherryPerry — APPROVED** [review]

**Pull request merged**